drivers/video/video_async.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "video_async.h"
+#include "video_ctrls.h"
+#include "video_device.h"
+
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/slist.h>
+
+LOG_MODULE_REGISTER(video_async, CONFIG_VIDEO_LOG_LEVEL);
+
+static sys_slist_t notifiers_list;
```

Comment:
I think this can be replaced by the [new device API](https://github.com/zephyrproject-rtos/zephyr/issues/82102) introduced just today.

Each of Zephyr `struct device *dev` is on a static array, and it is possible to get that array at runtime with [`z_device_get_all_static()`](https://github.com/zephyrproject-rtos/zephyr/blob/03482705f71257c5d8d86fcb6addc59bfc150a63/include/zephyr/device.h#L746).

Combined with this new API that allows to [check which API a device belongs to](https://github.com/zephyrproject-rtos/zephyr/blob/03482705f71257c5d8d86fcb6addc59bfc150a63/include/zephyr/device.h#L746), it is already possible to loop through all video devices.

Would that allow to remove some/all looping mechanisms and list to maintain?